internal/common/errors.go

```diff
@@ -32,18 +17,26 @@ func ValidationError(ctx *fiber.Ctx, errors []*ErrorResponse) error {
 }
 
 func CustomErrorHandler(ctx *fiber.Ctx, err error) error {
-	code := fiber.StatusInternalServerError
+	var problemJSON ProblemJSONError
 
-	if errors.Is(err, gorm.ErrRecordNotFound) {
-		return ClientErrorNotFound(ctx, err)
+	// Retrieve the custom status code if it's a fiber.*Error
+	var e *fiber.Error
```

Comment:
You will also need to implement the missing check for records not found that was present previously (if errors.Is(err, gorm.ErrRecordNotFound))

Reply:
Sorry, forgot it I see that you've updated the handlers